app/services/case_contacts_export_excel_service.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+# CaseContactsExportExcelService handles the conversion of case contact data to Excel format.
+# It provides methods for exporting case contact information to Excel files.
+require 'case_contacts_export_data_columns'
+
+class CaseContactsExportExcelService
+  attr_reader :case_contacts, :filtered_columns
+  include CaseContactsExportDataColumns
+
+  FONT_SIZE = 11
+  LINE_PADDING = 10
+
+  def initialize(case_contacts, filtered_columns = nil)
+    @filtered_columns = filtered_columns || CaseContactsExportDataColumns.data_columns.keys
+
+    @case_contacts = case_contacts.preload({ creator: :supervisor }, :contact_types, :casa_case)
+  end
+
+  def perform
+    Axlsx::Package.new do |p|
+      p.workbook.add_worksheet(name: 'Case Contacts') do |sheet|
+        sheet.add_row(filtered_columns.map(&:to_s).map(&:titleize))
+
+        if case_contacts.present?
+          wrap_style = p.workbook.styles.add_style(alignment: { wrap_text: true, vertical: :center })
+
+          case_contacts.decorate.each do |case_contact|
+            format_row(sheet, case_contact, wrap_style)
+          end
+          configure_case_contact_notes_width(sheet)
+          sheet.rows.each_with_index do |row, index|
+            infer_row_height(row) if index.positive?
+          end
+        end
+      end
+    end.to_stream.read
+  end
+
+  private
+
+  def configure_case_contact_notes_width(sheet)
+    case_contact_notes_index = filtered_columns.index(:case_contact_notes)
+
+    if case_contact_notes_index
```

Comment:
This was a good catch from the previous code. It was throwing a nil --> Integer error when the notes weren't a selected column.